src/tests/multihost/ad/test_adparameters_ported.py

```diff
@@ -0,0 +1,1749 @@
+          bz877972
+        :id: 9b71822b-09e0-48f9-9163-3b547364364e
+        :setup:
+         1. Configure ldap_sasl_authid to host/<HOSTNAME>@<AD_REALM>
+            clear cache and restart sssd.
+         2. Create an AD user.
+        :steps:
+          1. Run getent passwd for the user.
+          2. Run su for the user.
+          3. Check sssd domain log for expected messages:
+             Option ldap_sasl_authid has value host/<HOSTNAME>@<AD_REALM>
+             authid contains realm [<AD_REALM>]
+             Will look for host/<HOSTNAME>@<AD_REALM> in
+             Trying to find principal host/<HOSTNAME>@<AD_REALM> in keytab
+             Principal matched to the sample (host/<HOSTNAME>@<AD_REALM>)
+        :expectedresults:
+          1. User is found.
+          2. Su passes.
+          3. Expected lines are in the log.
+        :customerscenario: False
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=877972
+        """
+        ad_domain = multihost.ad[0].domainname
+
+        hostname_cmd = multihost.client[0].run_command(
+            'hostname -s',
+            raiseonerr=False
+        )
+        shortname = hostname_cmd.stdout_text.rstrip()
+        adjoin(membersw='adcli')
+        # Create AD user
+        (aduser, _) = create_aduser_group
+        # Configure sssd
+        ad_realm = multihost.ad[0].domainname.upper()
+        multihost.client[0].service_sssd('stop')
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        client.backup_sssd_conf()
+        dom_section = f'domain/{client.get_domain_section_name()}'
+        sssd_params = {
+            'debug_level': '9',
+            'ad_domain': multihost.ad[0].domainname.lower(),
+            'ad_server': multihost.ad[0].hostname,
+            'use_fully_qualified_names': 'False',
+            'cache_credentials': 'True',
+            #'ldap_id_mapping': 'False',
+            'ldap_sasl_authid': f'host/{shortname}.{ad_domain}@{ad_realm}',
+        }
+        client.sssd_conf(dom_section, sssd_params)
+        client.clear_sssd_cache()
+        # Search for the user
+        usr_cmd = multihost.client[0].run_command(
+            f'getent passwd {aduser}',
+            raiseonerr=False
+        )
+        # Run su command
+        su_cmd = multihost.client[0].run_command(
+            f'su - {aduser} -c  whoami',
+            raiseonerr=False
+        )
+        # Download the sssd domain log
+        log_str = multihost.client[0].get_file_contents(
+            f"/var/log/sssd/sssd_{multihost.ad[0].domainname.lower()}.log"). \
+            decode('utf-8')
+
+        # TODO: DELETE
+        multihost.client[0].run_command(
+            f"cat /var/log/sssd/sssd_{multihost.ad[0].domainname.lower()}.log",
+            raiseonerr=False
+        )
+
+        client.restore_sssd_conf()
+        client.clear_sssd_cache()
+
+        assert f"Option ldap_sasl_authid has value host/{shortname}.{ad_domain}@{ad_realm}" in log_str
```

Comment:
Are all the assertions failing?

Reply:
I did not manage to make the sasl tests work yet. (see @pytest.mark.broken instead of tier1).
I will look into it.